cumulusci/tasks/bulkdata.py

```diff
@@ -58,6 +59,22 @@ def setup_epoch(inspector, table, column_info):
         column_info["type"] = EpochType()
 
 
+class RelativeDateType(types.TypeDecorator):
+    impl = types.Integer
+
+    def __init__(self, now, *args, **kwargs):
+        self.now = now
+        super(RelativeDateType, self).__init__(*args, **kwargs)
+
+    def process_bind_param(self, value, dialect):
+        if isinstance(value, str):
+            value = datetime.datetime.fromisoformat(value[:-1])
+        return int((value - self.now).total_seconds())
+
+    def process_result_value(self, value, dialect):
+        return self.now + datetime.timedelta(seconds=value)
```

Comment:
This looks like it is more or less the same thing as EpochType except:
- it allows configuring the epoch date
- it coerces string bind params to datetimes
- it stores the offset in seconds rather than microseconds

Would you be okay switching it to store microseconds so we could merge the two classes? (Actually we could do that anyway if we added a `resolution` attribute.)

Reply:
I was actually confused about why EpochType uses microseconds anyhow.  The Salesforce DateTime field doesn't support microseconds.

Aside from that, the reason I did these as separate is that they have slightly different use cases:

* EpochType always uses the epoch date as the reference point and it seems clearer to have the class set that than to require it to be passed on init
* RelativeDateType requires you pass a value for now which is used to calculate the differences.  We could have a single class that optionally takes now as a kwarg but that seems a bit obscure.

Reply:
I'm not sure why it uses microseconds (that was here before I started working on it) but that is for the SQL db, not Salesforce.

They don't sound like different use cases to me. We're storing as an integer relative to a base date, which should be the standard epoch date by default but could be something else if specified. A single class whose `__init__` takes a `relative_to` param that defaults to the epoch date sounds perfect.